src/steinertree/steiner_tree.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    filter_non_term_leaves!(g, term_vert)
+
+Remove edges of `g` so that all non-isolated leaves of `g` are in the set `term_vert`
+"""
+function filter_non_term_leaves!(
+    g::AbstractGraph{T},
+    term_vert::Vector{<:Integer}
+    ) where T<:Integer
+
+    is_term = falses(nv(g))
+    is_term[term_vert] .= true
+    leaves = [v for v in vertices(g) if degree(g, v) == 1 && !is_term[v]]
+
+    while !isempty(leaves)
+        leaf = pop!(leaves)
+        for v in neighbors(g, leaf)
+            rem_edge!(g, v, leaf)
+            if degree(g, v) == 1 && !is_term[v]
+                push!(leaves, v)
+            end
+        end
+    end
+
+    return g
+end
+
+"""
+	steiner_tree(g, term_vert, distmx=weights(g))
+
+Return a minimum weight spanning graph of connected, undirected graph `g` with positive edge 
```

Comment:
Would it be possible to shorty (1 or two sentencs) describe here in the docstring, what steiner tree problem is? It is also not clear from the docstring, what `term_vert` is.

Reply:
Done